lib/src/compiler/linters.rs

```diff
@@ -114,6 +115,145 @@ impl LinterInternal for RuleName {
 
 type Predicate<'a> = dyn Fn(&Meta) -> bool + 'a;
 
+/// A linter that ensures tags meet specified requirements in either an allowed
+/// list of tags or in a regex.
+///
+/// ```
+/// # use yara_x::Compiler;
+/// use yara_x::linters;
+/// let mut compiler = Compiler::new();
+/// let warnings = compiler
+///     .add_linter(linters::tags_allowed(vec!["foo".to_string(), "bar".to_string()]))
+///     // This produces a warning because the rule tags are not from the
+///     // allowed list
+///     .add_source(r#"rule foo : test { strings: $foo = "foo" condition: $foo }"#)
+///     .unwrap()
+///     .warnings();
+///
+/// assert_eq!(
+///     warnings[0].to_string(),
+///     r#"warning[unknown_tag]: Tag not in allowed list
+///  --> line:1:12
+///   |
+/// 1 | rule foo : test { strings: $foo = "foo" condition: $foo }
+///   |            ---- tag `test` not in allowed list
+///   |
+///   = note: Allowed tags: foo, bar"#);
+pub struct Tags {
+    allow_list: Vec<String>,
+    regex: String,
+    compiled_regex: Option<Regex>,
+    error: bool,
+}
+
+impl Tags {
+    /// A list of strings that tags for each rule must match one of.
+    pub(crate) fn from_list(list: Vec<String>) -> Self {
+        Self {
+            allow_list: list,
+            regex: "".to_owned(),
+            compiled_regex: None,
+            error: false,
+        }
+    }
+
+    /// Regular expression that tags for each rule must match.
+    pub(crate) fn from_regex<R: Into<String>>(
+        regex: R,
+    ) -> Result<Self, regex::Error> {
+        let regex = regex.into();
+        let compiled_regex = Some(Regex::new(regex.as_str())?);
+        let tags = Self {
+            allow_list: Vec::new(),
+            regex,
+            compiled_regex,
+            error: false,
+        };
+        Ok(tags)
+    }
+
+    /// Specifies whether the linter should produce an error instead of a
+    /// warning.
+    ///
+    /// By default, the linter raises warnings about rule names that don't match
```

Comment:
The comment must be updated:  "By default, the linter raises warnings about *tags*..."